### PR TITLE
consolidate pin configuration in one file

### PR DESCRIPTION
**Pull request by creiter32**
Introduces an easy method to switch between TrustX shield on board
OPTIGA and TrustM2Go module.

### CODE REVIEW

File: pal/nrf5x/pal_pin_config.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+#define OPTIGA_PIN_ALL_MASKS            (OPTIGA_PIN_INITIAL_VAL_MASK | OPTIGA_PIN_ONE_TIME_INIT_MASK)
+
+#if OPTIGA_2GO == 1
+#define OPTIGA_PIN_VDD      (ARDUINO_9_PIN | OPTIGA_PIN_INITIAL_VAL_LOW | OPTIGA_PIN_ONE_TIME_INIT)
+#define OPTIGA_PIN_RST      (ARDUINO_7_PIN | OPTIGA_PIN_INITIAL_VAL_LOW)
+#elif  OPTIGA_TRUSTX_SHIELD == 1
+#define OPTIGA_PIN_VDD      (ARDUINO_9_PIN | OPTIGA_PIN_INITIAL_VAL_HIGH)
+#define OPTIGA_PIN_RST      (ARDUINO_7_PIN | OPTIGA_PIN_INITIAL_VAL_LOW)
+#endif
```
Comment (cmlesjak) [inline pal/nrf5x/pal_pin_config.h:81]: What if we add a final #else branch, and throw an error if none of the above if/elifs was true (i.e., == 1)?

**Review by cmlesjak — CHANGES_REQUESTED** [review]
Looks good! I have one idea to make the code more explicit, if no board is explicitly #defined. Can you check if this makes sense?

File: pal/nrf5x/pal_pin_config.h (hunk 1 above)
Comment (creiter32) [reply pal/nrf5x/pal_pin_config.h:81]: Good idea, implemented it.

**Review by cmlesjak — APPROVED** [review]
Thank you, looks good!

**Comment by cmlesjak** [comment]
PR accepted, looks good!

**Pull request merged**